xarray/backends/zarr.py

```diff
@@ -308,6 +308,17 @@ def _determine_zarr_chunks(
 
 
 def _get_zarr_dims_and_attrs(zarr_obj, dimension_key, try_nczarr):
+    # Zarr V3 explicitly stores the dimension names in the metadata
+    try:
+        # if this exists, we are looking at a Zarr V3 array
+        # convert None to empty tuple
+        dimensions = zarr_obj.metadata.dimension_names or ()
+        attributes = dict(zarr_obj.attrs)
+        return dimensions, attributes
+    except AttributeError:
+        # continue to old code path
+        pass
```

Comment:
`AttributeError` is a pretty broad exception, best to make it as narrowly scoped as possible, e.g.,
```suggestion
    try:
        # if this exists, we are looking at a Zarr V3 array
        metadata = zarr_obj.metadata
    except AttributeError:
        # continue to old code path
        pass
    else:
        # convert None to empty tuple
        dimensions = metadata.dimension_names or ()
        attributes = dict(zarr_obj.attrs)
        return dimensions, attributes
```